test/test_mixed_null_assertions.cpp

```diff
@@ -78,3 +78,207 @@ TEST(List_Mixed_do_insert)
     list.insert_null(0);
     list.insert(0, Mixed("hello world"));
 }
+
+TEST(Mixed_List_unresolved_as_null)
+{
+    Group g;
+    auto t = g.add_table("foo");
+    t->add_column_list(type_Mixed, "mixeds");
+    auto obj = t->create_object();
+    auto obj1 = t->create_object();
+
+    auto list = obj.get_list<Mixed>("mixeds");
+
+    list.insert_null(0);
+    list.insert(1, Mixed{"test"});
+    list.insert(2, obj1);
+    obj1.invalidate();
+
+    CHECK(list.size() == 3);
+
+    {
+        // find all mixed nulls or unresolved link should work the same way
+        std::vector<size_t> found;
+        std::vector<size_t> expected = {0, 2};
+        auto check_results = [&]() -> bool {
+            if (found.size() != expected.size())
+                return false;
+
+            std::sort(found.begin(), found.end());
+            std::sort(expected.begin(), expected.end());
+            for (size_t i = 0; i < found.size(); ++i) {
+                if (found[i] != expected[i])
+                    return false;
+            }
+            return true;
+        };
+        list.find_all(realm::null(), [&](size_t pos) {
+            found.push_back(pos);
+        });
+        CHECK(check_results());
+        found = {};
```

Comment:
You can make this a stronger test by changing the content of the expected values rather than checking that the results don't match the previously expected results.

Reply:
done